packages/spark/base/inputs/_text-icon-input.scss

```diff
@@ -12,6 +12,22 @@
   }
 }
 
+.sprk-b-TextInputIconContainer--has-text-icon-right {
```

Comment:
Hey @bobwise , we could make this code a little more extendable by breaking it out like this. Let me know your thoughts on this!

```
.sprk-b-TextInputIconContainer--has-text-icon,
.sprk-b-TextInputIconContainer--has-text-icon-right {
  &::before {
    content: $sprk-monetary-symbol;
    display: block;
    font-size: $sprk-input-text-icon-font-size;
    font-weight: $sprk-input-text-icon-font-weight;
    line-height: 1;
    margin-left: $sprk-input-text-icon-offset-x;
    position: absolute;
    top: calc(100% - #{$sprk-input-text-icon-offset-y});
    z-index: $sprk-input-text-icon-z-index;
  }
}

.sprk-b-TextInputIconContainer--has-text-icon-right {
  &::before {
    content: $sprk-monetary-symbol;
  }
}

.sprk-b-TextInputIconContainer--narrow {
  max-width: $sprk-input-container-narrow-max-width;
}

```

This allows the use of the new `sprk-b-TextInputIconContainer--narrow` class to apply the max-width setting you have. Having this allows apps to apply the narrow max-width if they want to other icon inputs or to remove that class if they are building their own textinput with a special icon.

For the other part, chaining the two selectors `.sprk-b-TextInputIconContainer--has-text-icon` and 
`.sprk-b-TextInputIconContainer--has-text-icon-right` allows the removal of all the duplicated ::before code to setup the icon.

Reply:
I added a new commit that consolidates all of the duplicated `::before` code. I think adding a new `sprk-b-TextInputIconContainer--narrow` class is scope creep. I'm going to be creating a new Issue to refactor this CSS to allow
- any input can be half-width, regardless of whether or not it has an icon
- inputs with an icon on the left can have any icon, not just $
- inputs with an icon on the right can have any icon, not just %

Reply:
@bobwise , thank you! Please link to the new issue here for future reference once you get that added.